src/main/kotlin/app/revanced/util/resources/ResourceUtils.kt

```diff
@@ -36,22 +49,116 @@ object ResourceUtils {
      *
      * @param sourceResourceDirectory The source resource directory name.
      * @param resources The resources to copy.
+     * @param replaceDestinationFiles If any existing destination files should be replaced.
+     *                                If set to false, an exception is thrown if any source file
+     *                                would overwrite an existing file.
      */
-    fun ResourceContext.copyResources(sourceResourceDirectory: String, vararg resources: ResourceGroup) {
+    fun ResourceContext.copyResources(
+        sourceResourceDirectory: String,
+        vararg resources: ResourceGroup,
+        replaceDestinationFiles: Boolean = true
+    ) {
         val classLoader = ResourceUtils.javaClass.classLoader
         val targetResourceDirectory = this["res"]
 
         for (resourceGroup in resources) {
             resourceGroup.resources.forEach { resource ->
+                // Create the target directory if it does not exist
+                this[targetResourceDirectory.resolve(resourceGroup.resourceDirectoryName).absolutePath].mkdir()
                 val resourceFile = "${resourceGroup.resourceDirectoryName}/$resource"
                 Files.copy(
                     classLoader.getResourceAsStream("$sourceResourceDirectory/$resourceFile")!!,
-                    targetResourceDirectory.resolve(resourceFile).toPath(), StandardCopyOption.REPLACE_EXISTING
+                    targetResourceDirectory.resolve(resourceFile).toPath(),
+                    *(if (replaceDestinationFiles) arrayOf(StandardCopyOption.REPLACE_EXISTING) else emptyArray())
                 )
             }
         }
     }
 
+    /**
+     * Copy localized strings from a given directory.
+     */
+    internal fun copyLocalizedStringFiles(context: ResourceContext, directory: String) {
+        val pattern = Pattern.compile("$directory/([-_a-zA-Z0-9]+)/strings\\.xml$")
+
+        var jf: JarFile? = null
+        try {
+            jf = JarFile(this.javaClass.protectionDomain.codeSource.location.toURI().path)
+            val entries = jf.entries()
```

Comment:
What we refer to "resource paths" or "resource directories" does not exist. Zip files do not have the semantic of folders and instead every folder we see in zip files is actually parsed from ZipEntry name. This means that every zip or jar file only has zip files and no folder structure. Because of that, jar resource file names should always be known at compile time whereas this method tries to retrieve the resource files at runtime dynamically. Instead, combining with my other comment regarding creating `values-<countrycode>` folders, and patches depending on the translations patch, the patch could iterate over each patch it registered, and try to load each resource file for every country code it knows (a list of country codes can dynamically be retrieved using `Locale.getISOCountries()`). The patch would try reading the resource, if it does not exist, try reading the resource file for the next country code, for each patch. This approach would spare relying on the approach seen in this method.